lib/datadog/tracing/contrib/sinatra/integration.rb

```diff
@@ -31,6 +31,14 @@ def new_configuration
             Configuration::Settings.new
           end
 
+          def subconfigure(tracing, sinatra)
```

Comment:
Lousy name, `configure` is already used.

I considered merely `def configure` and `super`ing but that requires giving arguments here that are sorta internal WRT this `Integration` DSL-ish descriptive class. So I preferred not leaking internal details.

Reply:
Maybe `instrument_dependencies` or `instrument_components`? It looks like in most cases this is what we end up doing here.

Reply:
Hmm.

Indeed in this case we do `c.instrument`, but that may not always be the case, it could also do `c.some_setting = :foo` in which case `instrument_*` is misleading.

Technically this is really a callback within the `Datadog.configure` block, so that we don't nest configure anymore. In that case I believe the name should include `configure` in some way for clarity. Since for `Integration`s there's a `def configure` that is called before, so maybe `def after_configure`? That would give clarity that i.e that it happens after `Integration#configure` has been called.

Functionally this is about setting up dependencies, so it may go beyond just `c.something = :foo` or `c.instrument`. So `def dependencies` or `def configure_dependencies` might cut it.

e.g I'll need to apply similar changes to `AppSec` so it makes me think that I'm yet unsure how that would unfold, therefore I'm trying to keep an open-ended approach where integrations could have dependencies across products. e.g `AppSec` or `CI` may decide that they'd want to enable tracing and/or enable specific tracing integrations when tracing is enabled to ensure Things Just Work.

Reply:
Good points. I like the `after_configure`, seems like it's a reasonable description for what's happening.